exporter/splunkhecexporter/metricdata_to_splunk.go

```diff
@@ -114,119 +151,43 @@ func metricDataToSplunk(logger *zap.Logger, data pdata.Metrics, config *Config)
-	for i := 0; i < len(bounds); i++ {
-		dimValue := float64ToDimValue(bounds[i])
-		splunkBounds[i] = &dimValue
-	}
-	splunkBounds[len(splunkBounds)-1] = &infinityBoundSFxDimValue
-
-	for i, bucket := range distributionValue.Buckets {
-		values[fmt.Sprintf("%s%s", bucketMetricName, *splunkBounds[i])] = bucket.Count
+func copy(fields map[string]interface{}) map[string]interface{} {
+	newFields := make(map[string]interface{}, len(fields))
+	for k, v := range fields {
+		newFields[k] = v
 	}
-
-	return values, nil
-}
-
-func float64ToDimValue(f float64) string {
-	// Parameters below are the same used by Prometheus
-	// see https://github.com/prometheus/common/blob/b5fe7d854c42dc7842e48d1ca58f60feae09d77b/expfmt/text_create.go#L450
-	// SignalFx agent uses a different pattern
-	// https://github.com/signalfx/signalfx-agent/blob/5779a3de0c9861fa07316fd11b3c4ff38c0d78f0/internal/monitors/prometheusexporter/conversion.go#L77
-	// The important issue here is consistency with the exporter, opting for the
-	// more common one used by Prometheus.
-	str := strconv.FormatFloat(f, 'g', -1, 64)
-	return str
+	return newFields
 }
 
-func mapSummaryValue(metric *metricspb.Metric, summaryValue *metricspb.SummaryValue) (map[string]interface{}, error) {
-
-	// Translating summary values per symmetrical recommendations to Prometheus:
-	// https://docs.signalfx.com/en/latest/integrations/agent/monitors/prometheus-exporter.html#overview
-
-	// 1. The total count gets converted to a cumulative counter called
-	// <basename>_count.
-	// 2. The total sum gets converted to a cumulative counter called <basename>
-	values := make(map[string]interface{})
-	metricName := fmt.Sprintf("%s:%s", splunkMetricValue, metric.GetMetricDescriptor().Name)
-	values[metricName+separator+countSuffix] = summaryValue.Count.Value
-	values[metricName] = summaryValue.Sum.Value
-
-	// 3. Each quantile value is converted to a gauge called <basename>_quantile
-	// and will include a dimension called quantile that specifies the quantile.
-	percentiles := summaryValue.GetSnapshot().GetPercentileValues()
-	if percentiles == nil {
-		return values, fmt.Errorf(
-			"unknown percentiles values for summary metric %q",
-			metricName)
+func timestampToEpochMilliseconds(ts pdata.TimestampUnixNano) *float64 {
+	if ts == 0 {
+		// some telemetry sources send data with timestamps set to 0 by design, as their original target destinations
+		// (i.e. before Open Telemetry) are setup with the know-how on how to consume them. In this case,
+		// we want to omit the time field when sending data to the Splunk HEC so that the HEC adds a timestamp
+		// at indexing time, which will be much more useful than a 0-epoch-time value.
+		return nil
 	}
-	quantileMetricName := metricName + separator + quantileSuffix + separator
-	for _, quantile := range percentiles {
-		values[quantileMetricName+float64ToDimValue(quantile.Percentile)] = quantile.Value
 
-	}
+	val := math.Round(float64(ts)/1e6) / 1e3
```

Comment:
Loss of precision of int64 here. Don't convert to float64 before division. float64 has only 53 bits of precision. This essentially throws away 11 bits of nanosecond precision, turning it to about 2-millisecond precision. This should avoid the problem:
```suggestion
	val := float64(ts/1e6) / 1e3
```

Reply:
No, that makes tests fail, as 32001999345 is not rounded to 32.002 but to 32.001 in that case.
Let me try to replicate what you're saying in a test case and see if I can verify it.

Reply:
I can't replicate. I pushed a test to show that I get a correct rounding with even a single nanosecond.